core/src/main/scala/org/apache/spark/rapids/tool/benchmarks/BenchmarkBase.scala

```diff
@@ -47,6 +51,69 @@ abstract class BenchmarkBase {
   def prefix: String = "rapids-tools-benchmark"
   def suffix: String = ""
 
+  /**
+   * Add a benchmark case to the suite
+   * @param name Name of the benchmark case
+   * @param numIters Number of iterations to run
+   * @param f Function to run
+   */
+  def addCase(name: String, numIters: Int = 0)(f: Int => Unit): Unit = {
+    benchmark.get.addCase(name, numIters)(f)
+  }
+
+  /**
+   * Method to trigger the benchmarker cases run method
+   */
+  def run(): Unit = {
+    // Getting correct output stream
+    val printStream = output.getOrElse(System.out)
+    // Running the underlying benchmark
+    val results: mutable.ArrayBuffer[Benchmark.Result] = benchmark.get.run()
+    // Generating the output report
+    val firstBest = results.head.bestMs
+    val nameLen = Math.max(40, results.map(_.caseName.length).max)
+    printStream.printf(s"%-${nameLen}s %14s %14s %11s %20s %18s %18s %18s %18s %10s\n",
+      "Benchmark :", "Best Time(ms)", "Avg Time(ms)", "Stdev(ms)","Avg GC Time(ms)",
+      "Avg GC Count", "Stdev GC Count","Max GC Time(ms)","Max GC Count", "Relative")
+    printStream.println("-" * (nameLen + 160))
+    results.foreach { result =>
+      printStream.printf(s"%-${nameLen}s %14s %14s %11s %20s %18s %18s %18s %18s %10s\n",
```

Comment:
We can use `f` interpolator to format the header string to include precision values:
```suggestion
      printStream.printf(f"%-${nameLen}s %14.0f %14.0f %11.0f %20.1f %18.1f %18.0f %18d %18d %10.2fX\n",
```

This would allow us to skip the redundant `"%5.0f" format` statements in all lines below